text/0000-otlp-http.md

```diff
@@ -0,0 +1,258 @@
+[OTLP](0035-opentelemetry-protocol.md) (OpenTelemetry Protocol).
+
+## Table of Contents
+
+* [Motivation](#motivation)
+* [OTLP/HTTP Protocol Details](#otlphttp-protocol-details)
+  * [Request](#request)
+  * [Response](#response)
+    * [Success](#success)
+    * [Failures](#failures)
+    * [Throttling](#throttling)
+    * [All Other Responses](#all-other-responses)
+  * [Connection](#connection)
+  * [Parallel Connections](#parallel-connections)
+* [Prior Art and Alternatives](#prior-art-and-alternatives)
+* [Performance Test Results](#performance-test-results)
+
+## Motivation
+
+OTLP can be currently communicated only via one transport: gRPC. While using
+gRPC has certain benefits there are also drawbacks:
+
+- Some users have infrastructure limitations that make gRPC-based protocol
+  usage impossible. For example AWS ALB does not support gRPC connections.
+
+- gRPC is a relatively big dependency, which some clients are not willing to
+  take. Plain HTTP is a smaller dependency and is built in the standard
+  libraries of many programming languages.
+
+## OTLP/HTTP Protocol Details
+
+This proposal keeps the existing specification of OTLP over gRPC transport
+(OTLP/gRPC for short) and defines an additional way to use OTLP protocol over
+HTTP transport (OTLP/HTTP for short). OTLP/HTTP uses the same ProtoBuf payload
+that is used by OTLP/gRPC and defines how this payload is communicated over HTTP
+transport.
+
+OTLP/HTTP uses HTTP POST requests to send telemetry data from clients to
+servers. Implementations MAY use HTTP/1.1 or HTTP/2 transports. Implementations
+that use HTTP/2 transport SHOULD fallback to HTTP/1.1 transport if HTTP/2
+connection cannot be established.
+
+### Request
+
+Telemetry data is sent via HTTP POST request. The request body is a
+ProtoBuf-encoded message `TelemetryRequest` as defined below:
+
+```protobuf
+enum RequestType {
+    _ = 0;
+    TraceExport = 1;
+    MetricExport = 2;
+}
+message TelemetryRequest {
+    RequestType request_type = 1;
+
+    // Only one of the following fields is supposed to contain data (determined
+    // by `request_type` field). This is deliberately not using ProtoBuf `oneof`
+    // for performance reasons (verified by benchmarks).
+    ExportTraceServiceRequest export_trace_request = 2;
+    ExportMetricsServiceRequest export_metrics_request = 3;
+}
+```
+
+`TelemetryRequest` references `ExportTraceServiceRequest` and
+`ExportMetricsServiceRequest` message types that are defined in
+[OTLP ProtoBuf schema](https://github.com/open-telemetry/opentelemetry-proto/tree/master/opentelemetry/proto)
+specification.
+
+The client MUST set "Content-Type: application/x-protobuf" request header. The
+client MAY gzip the content and in that case SHOULD include "Content-Encoding:
+gzip" request header. The client MAY include "Accept-Encoding: gzip" request
+header if it can receive gzip-encoded responses.
+
+The default request URL path is `/telemetry` (for example the full URL when
```

Comment:
Why not using two paths instead of an enum? It is easier to configure LBs I think

Reply:
That's a good point. It will definitely make routing the traffic easier.

Reply:
Changed to use URL path.